lib/schedule.js

```diff
@@ -188,40 +188,50 @@ Job.prototype.schedule = function(spec) {
   var inv;
   var start;
   var end;
+  var timezone = null;
 
   if (typeof spec === 'object' && spec.rule) {
     start = spec.start || null;
     end = spec.end || null;
+    timezone = spec.timezone || null;
     spec = spec.rule;
 
     if (start != null) {
-      if (!(start instanceof Date)) {
-        start = new Date(start);
+      if (!(start instanceof CronDate)) {
+        start = new CronDate(start, timezone);
       }
       if (!isValidDate(start) || start.getTime() < Date.now()) {
         start = null;
       }
+
     }
 
-    if (end != null && !(end instanceof Date) && !isValidDate(end = new Date(end))) {
+    if (end != null && !(end instanceof Date) && !(end instanceof CronDate)
+      && !isValidDate(end = new CronDate(end, timezone))) {
       end = null;
```

Comment:
This is difficult to read. I would change the style to something like this:
 ```js
if (end != null &&
     !(end instanceof Date) &&
     !(end instanceof CronDate) &&
     !isValidDate(end = new CronDate(end, timezone)) {
```
Also, I don't think the logic is (was) completely correct: we should check if `end` is a valid date whether the `end` was originally an instance of Date CronDate or no